assets/js/base/context/hooks/cart/use-store-cart-event-listeners.ts

```diff
@@ -2,25 +2,50 @@
  * External dependencies
  */
 import { useEffect } from '@wordpress/element';
-import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+import { CART_STORE_KEY } from '@woocommerce/block-data';
 import { dispatch } from '@wordpress/data';
 import { translateJQueryEventToNative } from '@woocommerce/base-utils';
 
 interface StoreCartListenersType {
+	// Counts the number of consumers of this hook so we can remove listeners when no longer needed.
 	count: number;
+	// Function to remove all registered listeners.
 	remove: () => void;
 }
 
+interface CartDataCustomEvent extends Event {
+	detail?:
+		| {
+				preserveCartData?: boolean | undefined;
+		  }
+		| undefined;
+}
+
 declare global {
 	interface Window {
 		wcBlocksStoreCartListeners: StoreCartListenersType;
 	}
 }
 
-const refreshData = ( e ): void => {
-	const eventDetail = e.detail;
+const refreshData = ( event: CartDataCustomEvent ): void => {
+	const eventDetail = event?.detail;
 	if ( ! eventDetail || ! eventDetail.preserveCartData ) {
-		dispatch( storeKey ).invalidateResolutionForStore();
+		dispatch( CART_STORE_KEY ).invalidateResolutionForStore();
+	}
+};
+
+/**
+ * Refreshes data if the pageshow event is triggered by the browser history.
+ *
+ * The deprecated performance object is needed in chrome since persisted is not reliable.
+ */
+const refreshDataIfPersisted = ( event: PageTransitionEvent ): void => {
+	if (
+		event?.persisted ||
+		( typeof window.performance !== undefined &&
+			window.performance.navigation.type === 2 )
```

Comment:
Just did some googling, wondering if we could avoid the deprecated code by using `window.performance.getEntriesByType("navigation")[0].type` instead, and checking if it equals `back_forward`? What do you think?

Reply:
Switched this now if you want to take another look.